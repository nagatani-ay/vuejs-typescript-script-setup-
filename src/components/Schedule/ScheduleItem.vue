<script setup lang="ts">
import { Todo } from '../../types';
import { ref } from 'vue';
import CustomButton from '../controls/Button.vue';
import TextInput from '../controls/TextInput.vue';
import DateInput from '../controls/DateInput.vue';
import { toStringDeadline, toObjectDeadline } from '../../Function/utils';
import { useTodos } from '../../Function/useTodos';

const props = defineProps<{ calanderTodo: Todo }>();

const isOpen = ref(false);
const tempText = ref(props.calanderTodo.text);
const tempDeadline = ref(toStringDeadline(props.calanderTodo.deadline));
function toggleMenu() {
  isOpen.value = !isOpen.value;
}

const { edit, editData } = useTodos();
function onEdit() {
  if (tempText.value === '' || tempDeadline.value === '') {
    alert('値を入力してください');
  } else {
    edit(editData(props.calanderTodo, tempText.value, tempDeadline.value));
    toggleMenu();
  }
}
</script>

<template>
  <div class="todo__item" @dblclick="toggleMenu()">
    <p>{{ calanderTodo.text }}</p>
    <div class="item__menu" v-if="isOpen">
      <custom-button @click="toggleMenu()">×</custom-button>
      <text-input v-model="tempText"></text-input>
      <date-input v-model="tempDeadline"></date-input>
      <custom-button @click="onEdit()">完了</custom-button>
    </div>
  </div>
</template>

<style>
.todo__item {
}
.item__menu {
  z-index: 2;
  position: absolute;

  background-color: white;
  border: solid 1px rgb(34 197 94);
}
</style>
