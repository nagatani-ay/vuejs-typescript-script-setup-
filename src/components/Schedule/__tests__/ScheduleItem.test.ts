import ScheduleItem from '../ScheduleItem.vue';
import { mount } from '@vue/test-utils';
import { describe, it } from 'vitest';

describe('ScheduleItem', () => {
  it('v-ifが表示されていないことの確認', async () => {
    const wrapper = mount(ScheduleItem, {
      props: {
        calanderTodo: {
          code: 'testData1',
          text: 'test1',
          status: false,
          time: 'time',
          deadline: {
            year: 2022,
            month: 5,
            day: 22,
          },
        },
      },
    });
    const div = wrapper.find('.todo__item');
    expect(wrapper.find('.item__menu').exists()).toBe(false);
  });

  it('ダブルクリックでv-ifの表示がされる', async () => {
    const wrapper = mount(ScheduleItem, {
      props: {
        calanderTodo: {
          code: 'testData1',
          text: 'test1',
          status: false,
          time: 'time',
          deadline: {
            year: 2022,
            month: 5,
            day: 22,
          },
        },
      },
    });
    const div = wrapper.find('.todo__item');
    await div.trigger('dblclick');
    expect(wrapper.find('.item__menu').exists()).toBe(true);
  });
});
