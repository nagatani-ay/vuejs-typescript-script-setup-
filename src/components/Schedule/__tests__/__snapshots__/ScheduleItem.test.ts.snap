// Vitest Snapshot v1
exports[`ScheduleItem > ダブルクリックによるv-ifの表示切替 > <div class="todo__item">
        <p>test1</p>
        <!--v-if-->
      </div> 1`] = `
"<div class=\\"todo__item\\">
  <p>test1</p>
  <!--v-if-->
</div>"
`;
